src/blob/core.rs

```diff
@@ -350,51 +370,62 @@ where
         key: &K,
         meta: Option<&Meta>,
         check_filters: bool,
-    ) -> Result<Option<Entry>> {
-        debug!("blob get any entry {:?}, {:?}", key, meta);
+    ) -> Result<ReadResult<Entry>> {
+        println!("blob get any entry {:?}, {:?}", key, meta);
         if check_filters && self.check_filter(key).await == FilterResult::NotContains {
-            debug!("Key was filtered out by filters");
-            Ok(None)
+            println!("Key was filtered out by filters");
+            Ok(ReadResult::NotFound)
         } else if let Some(meta) = meta {
-            debug!("blob get any entry meta: {:?}", meta);
+            println!("blob get any entry meta: {:?}", meta);
             self.get_entry_with_meta(key, meta).await
         } else {
-            debug!("blob get any entry bloom true no meta");
-            if let Some(header) = self.index.get_any(key).await.with_context(|| {
-                format!("index get any failed for blob: {:?}", self.name.to_path())
-            })? {
-                let entry = Entry::new(header, self.file.clone());
-                debug!("blob, get any entry, bloom true no meta, entry found");
-                Ok(Some(entry))
-            } else {
-                Ok(None)
-            }
+            println!("blob get any entry bloom true no meta");
+            Ok(self
+                .index
+                .get_any(key)
+                .await
+                .with_context(|| {
+                    format!("index get any failed for blob: {:?}", self.name.to_path())
+                })?
+                .map(|header| {
+                    let entry = Entry::new(header, self.file.clone());
+                    debug!("blob, get any entry, bloom true no meta, entry found");
+                    entry
+                }))
         }
     }
 
-    async fn get_entry_with_meta(&self, key: &K, meta: &Meta) -> Result<Option<Entry>> {
+    async fn get_entry_with_meta(&self, key: &K, meta: &Meta) -> Result<ReadResult<Entry>> {
         let headers = self.index.get_all(key).await?;
-        if let Some(headers) = headers {
-            let entries = Self::headers_to_entries(headers, &self.file);
-            self.filter_entries(entries, meta).await
-        } else {
-            Ok(None)
-        }
+        headers
+            .try_map_async(|headers| async move {
+                let entries = Self::headers_to_entries(headers, &self.file);
+                self.filter_entries(entries, meta).await
+            })
+            .await
     }
 
-    async fn filter_entries(&self, entries: Vec<Entry>, meta: &Meta) -> Result<Option<Entry>> {
+    async fn filter_entries(&self, mut entries: Vec<Entry>, meta: &Meta) -> Result<Entry> {
+        entries.reverse();
```

Comment:
I'm sorry, I was wrong: according by the latest changes in request #209, the actual order is from the newest entry to the oldest. So there is no need to reverse